frontend/source/js/app.js

```diff
@@ -0,0 +1,979 @@
+    // FILE DOWNLOAD
+    // =============================================================================
+
+    /**
+     * Downloads a CSV file from a protected API endpoint using a Bearer token.
+     * @async
+     * @param {number} year - The year of the data file to download
+     * @param {number} month - The month of the data file to download
+     * @param {string} token - The Bearer token used for authorization
+     * @returns {Promise<void>}
+     */
+    const downloadCsvFile = async (year, month, token) => {
+        const sensorData = selectedFeature?.get('sensorData');
+        const sensorId = sensorData?.id;
+
+        if (!sensorId) {
+            alert('No sensor selected');
+            return;
+        }
+
+        const url = `${CONFIG.API_BASE}/weatherstations/${sensorId}/files/${year}/${month}`;
+
+        try {
+            const response = await fetch(url, {
+                method: 'GET',
+                headers: {
+                    'Authorization': `Bearer ${token}`
+                }
+            });
+
+            if (!response.ok) {
+                throw new Error(`Download failed with status ${response.status}`);
+            }
+            // Validate Content-Type header
+            const contentType = response.headers.get('Content-Type');
+            if (!contentType || !contentType.includes('text/csv')) {
+                throw new Error('Invalid file type: Only CSV files are allowed');
+            }
+            const blob = await response.blob();
+            if (blob.type && !blob.type.includes('text/csv') && !blob.type.includes('application/csv')) {
+                throw new Error('Invalid file type: Only CSV files are allowed');
+            }
+
+            const downloadUrl = window.URL.createObjectURL(blob);
+
+            const link = document.createElement('a');
+            link.href = downloadUrl;
+            link.download = `${sensorId}_${year}_${month}.csv`;
+
+            document.body.appendChild(link);
+            link.click();
+            link.remove();
+
+            window.URL.revokeObjectURL(downloadUrl);
+        } catch (error) {
+            ErrorHandler.handle(error, 'download', 'Failed to download file');
+        }
+    };
+
+    // =============================================================================
+    // FILE RENDERING
+    // =============================================================================
+
+    /**
+     * Creates a download button for a file.
+     * @param {Object} fileData - File data with token, year, month
+     * @param {string} monthName - Display name for the month
+     * @returns {jQuery} Button element
+     */
+    const createDownloadButton = (fileData, monthName) => {
+        const $button = $('<button>')
+            .addClass('btn btn-link download-observation')
+            .attr({
+                type: 'button',
+                'data-file': fileData.token,
```

Comment:
So, every button has a scrapable and copy-pastable unencrypted token? 🗡️ 
Recommendations:
- Store in user memory temporarily with access via hash
- Generate this token upon request to a token-generating API and redirect